farmdata2_modules/fd2_tabs/resources/DateRangeSelectionComponent.spec.comp.js

```diff
@@ -73,14 +73,52 @@ describe('date range selection component', () => {
                     })
         })
     })
+
+    context('testing disabled option', () => {
+        let comp;
+        beforeEach(() => {
+            //ShallowMount to test disabled option
+            comp = shallowMount(DateRangeSelectionComponent, { 
+                propsData: {
+                    setStartDate: "2021-01-01",
+                    setEndDate: "2021-12-01",
+                    disabled: true,
+                },
+            })
+        })
+        
+        it('checking if the datePicker is disabled', () => {
+            mount(DateRangeSelectionComponent, {
+                propsData: {
+                    setStartDate: "2021-01-01",
+                    setEndDate: "2021-12-01",
+                    disabled: true,
+                }
+            })
+            cy.get('[data-cy=start-date-select]')
+            .get('[data-cy=date-select]')
+            .should('be.disabled')
+            cy.get('[data-cy=end-date-select]')
+            .get('[data-cy=date-select]')
+            .should('be.disabled')
+        })
+
+        it('checking disabled when prop changes', () => {
+            expect(comp.vm.isDisabled).to.equal(true)
+            cy.wrap(comp.setProps({ disabled: false })).then(() => {
+                expect(comp.vm.isDisabled).to.equal(false)
```

Comment:
This checks that the vue data is updated which is good.  The overall goal here is that the date elements should then be enabled.  So, it would be a little better to check for that here also.  You should add checks like you have above that check that that date elements are now enabled.

Reply:
Difficulty here is that in order to make  a prop change the component must be shallow mounted.  But when it is shallow mounted it is not possible to access the elements in the component.  So for now, it is necessary to just test the vue data model when testing watches for prop changes.  The E2E tests for pages that use the component and the props will inevitably test the functionality implemented by the watch and prop changes.  So we should still get coverage of that functionalilty.